### PR TITLE
Peft + finetuning fixes

### PR DESCRIPTION
**Pull request by JegernOUTT**

### CODE REVIEW

File: known_models_db/refact_known_models/huggingface.py [hunk 1]
```diff
@@ -280,7 +280,7 @@
         "chat_scratchpad_class": None,
         "model_class_kwargs": {},
         "required_memory_mb": 16000,
-        "T": 4096,
+        "T": 2048,
```
Comment (JegernOUTT) [inline known_models_db/refact_known_models/huggingface.py:283]: To fit in the 20gb gpu while training

File: refact_utils/finetune/train_defaults.py [hunk 2]
```diff
@@ -10,7 +10,6 @@
     "lr_decay_steps": 250,
     "lora_r": 16,
     "lora_alpha": 32,
-    "lora_init_scale": 0.01,
```
Comment (JegernOUTT) [inline refact_utils/finetune/train_defaults.py:13]: We should not configure `lora_init_scale` in the `peft` anymore

File: refact_webgui/webgui/static/tab-finetune.html [hunk 3]
```diff
@@ -268,10 +268,10 @@ <h5 class="modal-title" id="urlModalLabel">Finetune Settings</h5>
               <div class="col">
                 <div class="mb-3">
                   <label for="lora_r" class="form-label">Lora R</label>
-                  <input tabindex="-1" type="text" data-min="4" data-max="256" class="validate form-control" id="lora_r">
+                  <input tabindex="-1" type="text" data-min="4" data-max="64" class="validate form-control" id="lora_r">
```
Comment (JegernOUTT) [inline refact_webgui/webgui/static/tab-finetune.html:271]: correcting values considering vllm multilora limitations

File: self_hosting_machinery/finetune/configuration/supported_models.py [hunk 4]
```diff
@@ -58,7 +58,7 @@
     "train_model_modifiers": [
         "flash_sa.apply_flash_mha_to_starcoder2_model"
     ],
-    "force_enable_checkpointing": False
+    "force_enable_checkpointing": True
```
Comment (JegernOUTT) [inline self_hosting_machinery/finetune/configuration/supported_models.py:61]: otherwise starcoder2 models are too large while being in training

File: self_hosting_machinery/finetune/configuration/supported_models.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
 __all__ = ['config']
 
 _fim_train_ds_pipeline = {
-    "ds_opts": "n_ctx={n_ctx},debug=0,seed=42,shuffle_depth=256,"
+    "ds_opts": "n_ctx={n_ctx},debug=0,shuffle_depth=256,"
```
Comment (JegernOUTT) [inline self_hosting_machinery/finetune/configuration/supported_models.py:4]: we do not need fixed seed, 'cause it makes data per epochs the same
while multigpu training though it is necessary, so related changes would be in the next PR

File: self_hosting_machinery/finetune/scripts/auxiliary/file_sets_context.py [hunk 6]
```diff
@@ -37,6 +39,29 @@ def get_loss_by_content(self, model_name: str, content: str) -> Optional[float]:
         h = hashlib.sha1(content.encode("utf-8")).hexdigest()
         return self.loss_per_hash_db[(h, model_name)]["loss"] if (h, model_name) in self.loss_per_hash_db else None
 
+    def is_up_to_date(self) -> bool:
```
Comment (JegernOUTT) [inline self_hosting_machinery/finetune/scripts/auxiliary/file_sets_context.py:42]: get back `is_up_to_date` to prevent long finetune start when the files are already filtered

File: self_hosting_machinery/finetune/scripts/auxiliary/file_status_context.py [hunk 7]
```diff
@@ -4,7 +4,7 @@
 
 from refact_utils.scripts import env
 from refact_utils.finetune.utils import get_file_digest
-from self_hosting_machinery.finetune.scripts.aux.finetune_filter_status_tracker import FinetuneFilterStatusTracker
+from self_hosting_machinery.finetune.scripts.auxiliary.finetune_filter_status_tracker import FinetuneFilterStatusTracker
```
Comment (JegernOUTT) [inline self_hosting_machinery/finetune/scripts/auxiliary/file_status_context.py:7]: aux -> auxiliary to prevent the `aux` folder naming problem on windows machines

File: self_hosting_machinery/finetune/scripts/auxiliary/model.py [hunk 8]
```diff
@@ -121,29 +108,24 @@ def _make_model(
...
-        )
-        LoraMixin.apply_lora(
-            model.to(device),
-            lora_target_modules=lora_target_modules,
-            lora_r=int(lora_r),
-            lora_alpha=lora_alpha,
-            lora_dropout=lora_dropout,
-            lora_init_scale=lora_init_scale
-        )
+        if len(lora_target_modules) > 0:
```
Comment (JegernOUTT) [inline self_hosting_machinery/finetune/scripts/auxiliary/model.py:111]: peft support

File: self_hosting_machinery/finetune/scripts/finetune_filter.py [hunk 9]
```diff
@@ -55,30 +57,34 @@ def force_include_exclude_filter(
...
-        model_context: ModelContext,
+        finetune_cfg: Dict[str, Any],
         dataset_context: FileSetsContext,
         files_status_context: FilesStatusContext,
         status_tracker: FinetuneFilterStatusTracker,
         *,
         filter_loss_threshold
 ):
-    def _get_file_loss(file) -> float:
+    def _get_file_loss(model_context, file) -> Tuple[ModelContext, float]:
```
Comment (JegernOUTT) [inline self_hosting_machinery/finetune/scripts/finetune_filter.py:67]: model_context is creating lazily on demand now

**Review by mitya52 — APPROVED** [review]
